### DISCUSSION TITLE
KSQL statements support

### DISCUSSION THREAD

**Discussion by sripathiacharya** [provectus/kafka-ui#1075] [Q&A] [answered]
Hi Team,

The following statements fail with "422 Invalid sql" error. The statements work directly on KSQLDB using CLI. Is there a list of supported statements? I am using build 0.2.2-SNAPSHOT.

- INSERT INTO statements. https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/insert-into/
- CREATE TYPE statements. https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/create-type/

Thanks.

**Comment by Haarolean** [comment]
Hey, thanks for reaching out. We'll take a look at this!

**Reply by iliax** [reply]
Hi @sripathiacharya , current ksql execution support is more like PoC and not ready to be used in production. Only create, drop, list, show, describe statements supported. 
We have plans to provide full ksql functionality support in future releases (will add items into [roadmap](https://github.com/provectus/kafka-ui/projects/8) soon)

**Reply by Haarolean** [reply]
I'll convert this to an issue since it's a problem in fact.

**Comment by Haarolean** [comment, marked as answer]
KSQL execution has been drastically improved within #1109. If something does not work, please raise an issue.
